public/app/features/explore/Explore.tsx

```diff
@@ -111,12 +113,13 @@ export class Explore extends React.PureComponent<ExploreProps> {
   }
 
   componentDidMount() {
-    const { exploreId, urlState, initialized } = this.props;
-    const { datasource, queries, range = DEFAULT_RANGE, ui = DEFAULT_UI_STATE } = (urlState || {}) as ExploreUrlState;
+    const { exploreId, urlState, initialized, timeZone } = this.props;
+    const { datasource, queries, range: urlRange, ui = DEFAULT_UI_STATE } = (urlState || {}) as ExploreUrlState;
     const initialDatasource = datasource || store.get(LAST_USED_DATASOURCE_KEY);
     const initialQueries: DataQuery[] = ensureQueries(queries);
-    const initialRange = { from: parseTime(range.from), to: parseTime(range.to) };
+    const initialRange = urlRange ? timeRangeFromUrlSelector(urlRange, timeZone).raw : DEFAULT_RANGE;
```

Comment:
I think you could do it on mapStateToProps level rather than here. Especially that you have all necessary information there already.

Reply:
It's done